### PR TITLE
release: Update release.md to reflect the schema that we settled on

### PR DESCRIPTION
**Pull request by fabiand**
After a few iterations we settled on a schema that is centered around the Kubernetes release date. This update is trying to document the guiding principles and documents the release schema.

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

**What this PR does / why we need it**:

Updates to the KubeVirt release schema.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fabiand** [comment]
@rthallisey @dhiller @davidvossel FYI

**Comment by fabiand** [comment]
@rthallisey please review

Please note that I extended the delta between kube and kubevirt release to 2 months. This is to ensure that we have capacity to stabilize the release.

No further request from my side, I'm ready to sign off

**Comment by fabiand** [comment]
@dhiller FYI

**Comment by fabiand** [comment]
@rmohr do you mind approving this PR?
Alternatively @aburdenthehand @davidvossel 

FWIW the wiki page got updated to reflect this new schedule https://github.com/kubevirt/kubevirt/wiki/Release-v0.59

File: docs/release.md [hunk 1]
```diff
@@ -39,79 +21,61 @@ These artifacts are provided in their respective channels and their natural way
...
+The assumption is, that the KubeVirt CI provider will be available by KubeVirt's enhcanement freeze.
+
+## Release schedule schema  
+
+This can then be translated into the following KubeVirt release schedule schema:
+
+| Week Number in Kubernetes Release Cycle | KubeVirt Milestone | Note                            |
+|:---------------------------------------:|--------------------|---------------------------------|
+| 14 - 8                                  | Alpha              | KubeVirt alpha.0 on main        |
+| 14 - 4                                  | Beta               | KubeVirt beta.0 on main         |
```
Comment (davidvossel) [inline docs/release.md:53]: I know this already merged. And i think it's definitely a good revision.

One think I might want to iterate on in the future is the cadence for alpha releases. I'd us to keep making some sort of release out every month even it is super early in the dev cycle.

**Pull request merged**